src/main/java/it/pagopa/pn/delivery/svc/search/NotificationSearchMultiPage.java

```diff
@@ -178,18 +121,49 @@ else if (indexNameAndPartitions.getIndexName().equals(IndexNameAndPartitions.Sea
+        PnLastEvaluatedKey pageLastEvaluatedKey = new PnLastEvaluatedKey();
+
+        if (indexNameAndPartitions.getIndexName().equals(IndexNameAndPartitions.SearchIndexEnum.INDEX_BY_SENDER))
+        {
+            pageLastEvaluatedKey.setExternalLastEvaluatedKey(keyelement.getSenderIdCreationMonth());
+            pageLastEvaluatedKey.setInternalLastEvaluatedKey(Map.of(
+                    NotificationMetadataEntity.FIELD_SENDER_ID_CREATION_MONTH, AttributeValue.builder().s(keyelement.getSenderIdCreationMonth()).build(),
+                    NotificationMetadataEntity.FIELD_SENT_AT, AttributeValue.builder().s(keyelement.getSentAt().toString()).build(),
+                    NotificationMetadataEntity.FIELD_IUN_RECIPIENT_ID, AttributeValue.builder().s(keyelement.getIunRecipientId()).build()));
+        }
+        else if (indexNameAndPartitions.getIndexName().equals(IndexNameAndPartitions.SearchIndexEnum.INDEX_BY_RECEIVER))
+        {
+            pageLastEvaluatedKey.setExternalLastEvaluatedKey(keyelement.getRecipientIdCreationMonth());
+            pageLastEvaluatedKey.setInternalLastEvaluatedKey(Map.of(
+                    NotificationMetadataEntity.FIELD_RECIPIENT_ID_CREATION_MONTH, AttributeValue.builder().s(keyelement.getRecipientIdCreationMonth()).build(),
+                    NotificationMetadataEntity.FIELD_SENT_AT, AttributeValue.builder().s(keyelement.getSentAt().toString()).build(),
+                    NotificationMetadataEntity.FIELD_IUN_RECIPIENT_ID, AttributeValue.builder().s(keyelement.getIunRecipientId()).build()));
+        }  else if (indexNameAndPartitions.getIndexName().equals(IndexNameAndPartitions.SearchIndexEnum.INDEX_WITH_BOTH_IDS))
+        {
+            pageLastEvaluatedKey.setExternalLastEvaluatedKey(keyelement.getSenderIdRecipientId());
+            pageLastEvaluatedKey.setInternalLastEvaluatedKey(Map.of(
+                    NotificationMetadataEntity.FIELD_SENDER_ID_RECIPIENT_ID, AttributeValue.builder().s(keyelement.getSenderIdRecipientId()).build(),
+                    NotificationMetadataEntity.FIELD_SENT_AT, AttributeValue.builder().s(keyelement.getSentAt().toString()).build(),
+                    NotificationMetadataEntity.FIELD_IUN_RECIPIENT_ID, AttributeValue.builder().s(keyelement.getIunRecipientId()).build()));
+        }
+        return pageLastEvaluatedKey;
+    }
+
     /**
      * Recupero tutti i dati da una partizione (ricorsivamente)
+     * @param currentRequest indice inizio scansione partizione
      * @param partition partizione di ricerca
      * @param cumulativeQueryResult risultati cumulativi da aggiornare
      * @param lastEvaluatedKey ultimo elemento letto nella partizione
      * @param requiredSize dimensione totale richiesta
      * @param dynamoDbPageSize dimensione della pagina da leggere in dynamo
-     * @return true se ci sono altri dati da cercare, false altrimenti
+     * @return numero di elementi letti
```

Comment:
ritorna il "numero di query eseguite" in realtà, non il numero di elementi letti